### DISCUSSION TITLE
Stream from DJI Mavic 2 Enterprise to mediamtx using RTMP

### DISCUSSION THREAD

**Discussion by mzpgdtm** [bluenviron/mediamtx#1673] [Questions]
Hello,

I would like to configure to stream the video from DJI Mavic 2 Enterprise drone using RTMP but I'm getting this error:
closed (unsupported video codec: 11)

I guess DJI drone is using unsupported codec.

How can I solve this?

Thank you!

**Comment by aler9** [comment]
<s>Hello, i checked in the RTMP codec list, and codec ID 11 is associated with the Speex codec:
https://github.com/FFmpeg/FFmpeg/blob/870bfe16a12bf09dca3a4ae27ef6f81a2de80c40/libavformat/flv.h#L101

I've created a feature request here: #1674

In the meanwhile, you have to disable audio or switch to AAC.</s>

**Comment by mzpgdtm** [comment]
Hello,

thank you for quick response. How can I switch to AAC or disable the audio?

Thank you!

**Reply by Miliks** [reply]
Hello, I'm facing the same issue and was wondering if u r managed to fix it?

**Reply by mzpgdtm** [reply]
Unfortunately not, I'm still wainting for the solution.

Best wishes
Marko

**Reply by aler9** [reply]
Hello, the question is that codec 11 is not part of the RTMP standard, therefore unless i put my hands on a drone there's no way to know how to parse that codec and read messages.

**Reply by Miliks** [reply]
Could you be more specific? I'm not using Mavic but Matrice, but i guess the SW is the same. When i chose the streaming there is no way to specify codec neither for audio nor for video. And the only standard URL allowed is RTMP, so i expect whet it is complaint to the formats supported by this protocol, isn't it as well as your server?

**Reply by aler9** [reply]
I just don't know how to parse codec 11 since there is no documentation about it anywhere. For instance, this is the list of RTMP/FLV codec IDs supported by FFmpeg:

https://ffmpeg.org/doxygen/trunk/flv_8h_source.html#l00104

As you can see, there is no codec ID with a value of 11.

If I had a drone or at least a software able to read that RTMP stream I'd be able to reverse-engineer it and add support for it into the server, otherwise it's like walking in the dark.

**Reply by Tokolosh** [reply]
@aler9 the latest restreamer 2.6.0 (the only version i tested with a mavic air 2) works fine with a dji drone rtmp stream.
In my case it just needed the rtmp address to be in this format   rtmp://rtmpserverurl:1935/xyz/token
Maybe that could help you. Here on github:  datarhei/restreamer

**Reply by Miliks** [reply]
@aler9. I think the number is random, i get different numbers trying to execute it. Sometimes it come with 2, 9, 11... so that is why i haven't associate it with the codec number. 
@Tokolosh: i don't think the problem is a port number. I can see from the media server log what it try to establish connection with the drone media stream, and it fails or with timeout or with the unsupported codec, and my guess is what is the port was necessary it wouldn't establish the connection. Correct me if i'm wrong.
Any suggestion on how i can discover the type of codec?

**Reply by Tokolosh** [reply]
@Miliks i was replying to @aler9 's post regarding:
[quote]If I had a drone or at least a software able to read that RTMP stream I'd be able to reverse-engineer it[/quote]
as datarhei restreamer works with dji rtmp streams just that i had to add the port to get it to work there

**Reply by aler9** [reply]
I checked Datarhei Restreamer and it's a clear example of "false open open source", since it seems open source but no source code is available. Anyway it can't be used as reference to implement this feature.

[a comment by an automated account is not shown]
